lib/indexer.rb

```diff
@@ -0,0 +1,151 @@
+module Argo
+  class SimpleLogJob
+    def initialize(log_val)
+      @log_val = log_val
+    end
+
+    def perform
+      sleep(rand 10)
+
+      log_filename = 'tmp/simple_job.log'
+      File.open(log_filename, 'a') { |log|
+        log.puts "#{@log_val}"
+      }
+    end
+  end
+
+  # TODO: DRY up the repetition btwn here and dor_controller
+  class Indexer
+    @@index_logger = nil
+    def self.index_logger
+      return @@index_logger if @@index_logger
+
+      @@index_logger ||= Logger.new("#{Rails.root}/log/indexer.log", 10, 3240000)
```

Comment:
after a cursory look into "class instance variables":  they seem like needless complexity for this case.  my impression is that they're used when you want a static variable that's shared among classes, but not super-classes or sub-classes (and i guess class variables are shared by sub-classes also).  i vote for leaving this as-is for now and dealing with this when `Indexer` gets subclassed and the subclass needs its own logger.

i'd also vote for turning off this rule.